ZEngine/src/ImguiLayer.cpp

```diff
@@ -63,10 +68,18 @@ namespace ZEngine::Layers
             ImGui_ImplGlfw_InitForVulkan(reinterpret_cast<GLFWwindow*>(current_window_ptr->GetNativeWindow()), false);
 
             // Create Descriptor Pool
-            std::vector<VkDescriptorPoolSize> pool_sizes = {{VK_DESCRIPTOR_TYPE_SAMPLER, 1000}, {VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, 1000},
-                {VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE, 1000}, {VK_DESCRIPTOR_TYPE_STORAGE_IMAGE, 1000}, {VK_DESCRIPTOR_TYPE_UNIFORM_TEXEL_BUFFER, 1000},
-                {VK_DESCRIPTOR_TYPE_STORAGE_TEXEL_BUFFER, 1000}, {VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1000}, {VK_DESCRIPTOR_TYPE_STORAGE_BUFFER, 1000},
-                {VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER_DYNAMIC, 1000}, {VK_DESCRIPTOR_TYPE_STORAGE_BUFFER_DYNAMIC, 1000}, {VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT, 1000}};
+            std::vector<VkDescriptorPoolSize> pool_sizes = {
```

Comment:
You can do it like this for the moment to get going with your implementation. Just be aware that creating a pool reserving n of all descriptor types might waste a lot of memory depending on the descriptor sets' layouts you allocate from that pool. 

For example, consider the case where you allocate 1000 descriptor sets with a layout containing only one sampled image. The descriptor pool will then not be able to allocate more descriptor sets of that layout or any layout containing sampled images. But the rest of the descriptor types are still reserved, so you have 10000 individual descriptors still reserved by that pool without using them.

Reply:
Thanks a lot! , Fixed now
This was coming from the ImGui example.